benchmarks/hia/countdown/Main.hs

```diff
@@ -0,0 +1,44 @@
+{- Inversion of control -}
+ 
+{-# LANGUAGE GADTs, TypeFamilies, MultiParamTypeClasses,
+    FlexibleInstances, FlexibleContexts, QuasiQuotes, DeriveFoldable,
+    BangPatterns #-}
+
+import System.Environment
+import HIA.Handlers
+import HIA.TopLevel
+import HIA.DesugarHandlers
+
+[operation|Get s :: s |]
+[operation|Put s :: s -> ()|]
+
+[handler|
+    RunState s a :: s -> (a, s) 
+    handles {Get s,Put s} where
+        Return x   s -> (x,s)
+        Get      k s -> (k s) s
+        Put    s k _ -> (k ()) s 
+|]
+
+type CountdownComp = Comp (RunState Int Int) Int
+
+countdown :: CountdownComp
+countdown = do
+        y <- get
+        if y == 0 then return y
+        else do
+            put (y - 1)
+            countdown
+
+run :: Int -> Int
+run n = fst (runState n countdown)
```

Comment:
Reading the benchmark description, it technically asks for the final state value (which happens to coincide with the return of `countdown` in this instance). It may be technically more correct, in some sense, to use `execState` (see my comment attached to `runState`).

Reply:
@dhil Together with @jasigal, we looked over the other implementations for countdown and they all seem to return the value instead of the state. You are right though, the benchmark description does require returning the state. The question is then, do we change the benchmark description or update all the implementations? Does it make a difference for performance whether the state or the value is returned?

Reply:
We should definitely update the description to ask for `runState`, `execState`, or `evalState`. I think it may be more fair/minimal to ask for one of the latter two, as they don't require returning multiple values...

Reply:
I agree! Should we maybe go for `evalState` instead of `execState` then? Since that is what the other languages seem to have implemented.

Reply:
I don't have strong opinions on either, so I am happy to go with `evalState` and a corresponding update to the benchmark description (should probably be done in a separate PR).

Reply:
Yes, I agree